### ISSUE TITLE
refactor(research): consolidate and isolate model evaluation pipeline

### ISSUE THREAD

**Issue by mauriziokeningen** [mauriziokeningen/Computer-Vision-Residential-Security-System#47]
# Refactor: Reorganize AI Research & Model Evaluation Pipeline

**Labels:** `refactor`, `architecture`, `MLOps`, `tech-debt`, `blocker`  
**Status:**  **High Priority** (Pre-requisite for the 2-week Movement Detection Sprint)

---

### **Problem Statement**
The current repository structure conflates production service logic with experimental research assets. Research code, heavy datasets, and localized virtual environments currently reside within the application folders. This results in significant technical debt:
* **Dependency Bloat:** Production environments are forced to carry heavy AI libraries (e.g., PyTorch, CUDA, Ultralytics) that are unnecessary for API service.
* **Data Locality Issues:** Heavy datasets and model weights are buried in nested directories, hindering reuse across different model architectures.
* **Metric Drift:** Redundant scripts and scattered results folders make benchmarking unreliable and difficult to replicate across the team.

### **The Solution: Research-to-Prod Isolation**
By establishing an isolated `/research` directory, the team can iterate on **Action Recognition**, **Pose Estimation**, and **Object Detection** models using heavy dependencies and massive datasets without affecting the stability of the core backend. Production environments will only receive "Golden" weights and optimized inference scripts exported from this laboratory environment.

---

### Phase 1: The Research Scaffold
Establish a standardized laboratory hierarchy to house all experimental assets:
* **Directory Creation:** Initialize top-level folders for `data`, `models`, `core`, and `notebooks`.
* **Domain Relocation:** Move all action recognition logic, skeleton parsers, and pose estimation scripts into the new hierarchy.
* **Weights & Artifacts:** Relocate all model weights (e.g., `.pt`, `.onnx`) to dedicated, version-controlled sub-folders within the models directory.
* **Dataset Centralization:** Consolidate all raw datasets and annotated skeleton data into a centralized data store.

### Phase 2: Dependency & Binary Management
* **Environment Clean-up:** Purge all local virtual environment directories from the repository history and ensure they are strictly ignored in the root configuration.
* **Binary Hygiene:** Ensure heavy model weights and binary artifacts are managed via large-file tracking or excluded from standard Git tracking to prevent repository bloat.
* **Requirement Standardization:** Consolidate AI-specific libraries into a dedicated research requirements file to ensure environment parity across the team.
* **Metric Unification:** Merge redundant evaluation scripts into a single, high-fidelity module. This module must output standardized performance metrics: $F_1 \text{ Score}$, $mAP$, and Confusion Matrices.

### Phase 3: Path and Import Correction
* **Import Resolution:** Update all internal script imports to reflect the new directory structure, moving toward a standardized package-based approach.
* **Asset Mapping:** Update hardcoded paths in the training and inference scripts to point to the new centralized data and weights directories.

---

## Definition of Done (DoD)

- [ ] **Physical Isolation:** The application directories contain zero datasets, training scripts, or research-only weights.
- [ ] **Git Hygiene:** Local environments are purged from history and correctly ignored in the root configuration.
- [ ] **Data & Weight Integrity:** Heavy datasets and training artifacts are excluded from standard tracking; weights are managed via optimized binary storage solutions.
- [ ] **Standardized Benchmarking:** A single source of truth exists for model evaluation across all research domains.
- [ ] **Target Architecture Blueprint:** The final directory structure must strictly adhere to the following blueprint:

```text
research/
├── data/                         # Centralized Data Store
│   ├── raw/                      # Original, immutable datasets
│   └── processed/                # Cleaned/Annotated data for training
├── models/                       # Domain-Specific Models
│   ├── movement_detection/       
│   │   ├── train.py
│   │   └── weights/              # Local weight storage (.pt, .onnx)
│   ├── facial_recognition/
│   │   └── weights/
│   └── object_detection/
│       └── weights/
├── core/                         # Shared Research Utilities
│   ├── evaluation.py             # Global metrics (F1, mAP, Confusion Matrix)
│   ├── preprocessing.py          # Data augmentation and parsing logic
│   └── inference_client.py       # Standardized wrapper for local testing
├── notebooks/                    # Experimental Jupyters/Playgrounds
└── requirements.txt              # AI-specific dependencies (PyTorch, etc.)
```
